hs_core/task_utils.py

```diff
@@ -28,15 +39,224 @@ def _retrieve_task_id(res_id, job_dict):
+            if username:
+                obj.username = username
+            if name:
+                obj.name = name
+            if payload:
+                obj.payload = payload
+            obj.status = status
+            obj.save()
+
+
+def get_resource_bag_task(res_id):
+    job_name = 'hs_core.tasks.create_bag_by_irods'
+    return _retrieve_job_id(job_name, res_id)
+
+
+def get_resource_delete_task(res_id):
+    job_name = 'hs_core.tasks.delete_resource_task'
+    return _retrieve_job_id(job_name, res_id)
+
+
+def get_all_tasks(username):
+    """
+    get all tasks by a user identified by username input parameter
+    :param username: the user to retrieve all tasks for
+    :return: list of tasks where each task is a dict with id, name, and status keys
+    """
+    i = inspect()
+    act_task_list, act_task_ids = _retrieve_user_tasks(username, i.active(), 'active')
+    res_task_list, res_task_ids = _retrieve_user_tasks(username, i.reserved(), 'reserved')
+    sched_task_list, sched_task_ids = _retrieve_user_tasks(username, i.scheduled(), 'scheduled')
+    task_list = act_task_list + res_task_list + sched_task_list
+    task_ids = act_task_ids.union(res_task_ids).union(sched_task_ids)
+    task_notif_list = []
+    for obj in TaskNotification.objects.all():
+        task_notif_list.append({
+            'id': obj.task_id,
+            'name': obj.name,
+            'status': dict(TaskNotification.TASK_STATUS_CHOICES)[obj.status],
+            'payload': obj.payload
+        })
+    task_list.extend(item for item in task_notif_list if item['id'] not in task_ids)
+    return task_list
+
+
+def get_task_by_id(task_id, name='', payload='', request=None):
+    """
+    get task dict by celery task id
+    :param task_id: task id
+    :param name: task name with default being empty
+    :param payload: task payload to use. If empty, task return result will be used as payload
+    :param request: the request object
+    :return: task dict with keys id, name, status
+    """
+    result = AsyncResult(task_id)
+    status = dict(TaskNotification.TASK_STATUS_CHOICES)['progress']
+    username = request.user.username if request else ''
+    if result.ready():
+        try:
+            ret_value = result.get()
+            status = dict(TaskNotification.TASK_STATUS_CHOICES)['completed']
+            if not payload:
+                payload = ret_value
+            if name == "resource delete" and request:
+                res = get_resource_by_shortkey(ret_value)
+                res_title = res.metadata.title
+                res_type = res.resource_type
+                post_delete_resource.send(sender=type(res), request=request, user=request.user,
+                                          resource_shortkey=ret_value, resource=res,
+                                          resource_title=res_title, resource_type=res_type)
+            create_task_notification(task_id=task_id, status=status, name=name, payload=payload, username=username)
```

Comment:
same with this, I think we should hook into a task success signal and then create the notification there.